Sources/Frame.swift

```diff
@@ -36,3 +40,52 @@ struct Frame: Equatable, Hashable {
     }
 
 }
+
+private func computeViewport(from image: NSImage) -> NSRect {
+    let cgImage = image.cgImage(forProposedRect: nil, context: nil, hints: nil)!
+    let representation = NSBitmapImageRep(cgImage: cgImage)
+    let centerX = Int(image.size.width / 2)
+    let centerY = Int(image.size.height / 2)
+    let centerColor = representation.colorAt(x: centerX, y: centerY)
+    let x1 = computeHorizontalStartOf(centerColor, in: representation, startingFromX: centerX, y: centerY)
+    let x2 = computeHorizontalEndOf(centerColor, in: representation, startingFromX: centerX, y: centerY)
+    let y1 = computeVerticalStartOf(centerColor, in: representation, startingFromX: centerX, y: centerY)
+    let y2 = computeVerticalEndOf(centerColor, in: representation, startingFromX: centerX, y: centerY)
+    return NSRect(x: x1, y: y1, width: x2 - x1 + 1, height: y2 - y1 + 1)
+}
+
+private func computeHorizontalStartOf(_ color: NSColor?, in representation: NSBitmapImageRep, startingFromX x: Int, y: Int) -> Int {
```

Comment:
There's a lot of repeated code here that differs by a tiny bit, not sure if there's an easy way to factor out or generalize.